### PR TITLE
Started refactoring the database helper.

### PR DESCRIPTION
**Pull request by dibari**
Conventions will have to be discussed.
IAMA-1892

I opened this PR to be able to comment on the code - you can't do it otherwise and I don't want comments to get lost in slack

### CODE REVIEW

File: lib/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private constructor(private readonly client: Client) { }
+
+    private static discardResult = () => { };
+
+    /**
+     * Executes a database query
+     * @param text SQL text
+     * @param values Parameter values
+     */
+    query(text: string, values: any[]): Promise<QueryResult> {
```
Comment (dibari) [inline lib/transaction.ts]: For queries that don't involve transactions, the user will have to reference transaction.query - is there an argument to pull out the query method into a separate class?

Comment (pcwiek) [inline lib/transaction.ts]: Although, to be fair, I'll have to add an overload with `query(sql : QueryStatement)` (or whatever that type was) to be able to use it freely with `sql-template-strings` package.

Comment (pcwiek) [inline lib/transaction.ts]: @dibari Now that I think about it, I'll remove the `query` method from the `Transaction` object altogether. It should only open and close the transaction, but the queries can be done using the `Pool`, so we'd have:
```
const db = new Pool();
const transaction = await db.beginTransaction();
try {
  await db.query(...); // this one is in transaction that has been opened above
  await db.query(...); // this one too
  await transaction.commit();
} catch(err) {
  await transaction.rollback();
}
await db.query(...); // this one is not
```

@dibari @CaptainJeff @Jaskar14 thoughts? :)

**Comment by pcwiek** [comment]
@CaptainJeff any thoughts on the `Transaction` wrapper?

Anyway, if we're OK with that, please approve the CR and we'll be able to merge it then.

**Pull request merged**